Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -323,30 +533,67 @@ extension ComponentAPICoordinator: NoResultsScreenDelegate {
     }
 }
 
+// MARK: - Validation
+
+extension ComponentAPICoordinator {
+    fileprivate func validate(_ documents: [GiniVisionDocument],
+                              completion: @escaping (Result<[DocumentRequest]>) -> Void) {
+        
+        guard !(documents + documentRequests.map {$0.document}).containsDifferentTypes else {
```

Comment:
In this method, why do you have all three stages inline? It's difficult to read. Just have three methods:
* checking for max pages
* file type
* custom validation

Reply:
I know about this and it is not my main concern here. My idea is to simplify this for the component API so clients wouldn't have to implement all this logic (it is copied from the ScreenAPICoordinator). 
But it is not an easy task since the `validation` method must be quite clear (you have to pass the validated documents collection and new documents). I'm planning to tackle this in another PR so I won't change anything here.